daliuge-engine/dlg/deploy/create_dlg_job.py

```diff
@@ -662,14 +697,12 @@ def main():
             num_islands=opts.num_islands,
             all_nics=opts.all_nics,
             check_with_session=opts.check_with_session,
-            logical_graph=opts.logical_graph,
-            physical_graph=opts.physical_graph,
+            logical_graph=lg_graph,
```

Comment:
This variable doesn´t appear to be in scope or contain anything once we get to this point in the code. Is it necessary?

Reply:
In this current way of doing things that is correct, but we could actually pass the logical graph to the client as well, which is what we might need to do for certain deployments. That would then add another case to the opts...

Reply:
There is still the issue of the `lg_graph` being possibly un-initialised as it is only declared/initialised in the `if opts.logical_graph` block. This can be resolved by moving the empty-initialisation currently on line 685 to just after the `else:` statement.